### PR TITLE
Implement product cards

### PR DESCRIPTION
**Pull request by Mana12011207**
Hi, I implemented new arrivals and featured collections on the home page.

### CODE REVIEW

File: src/pages/Home.jsx [hunk 1]
```diff
@@ -1,9 +1,134 @@
...
+          <h1 className='text-left font-bold text-3xl mt-10 mb-4 text-primary'>New Arrivals</h1>
+          <div className='carousel mt-4 overflow-hidden'>
+            <motion.div
+              className='flex space-x-2'
+              drag='x'
+              dragConstraints={{ left: -1000, right: 0 }}
+            >
+              {/* Iterates over each item in the newArrivals array. */}
+              {newArrivals.map((product, index) => (
+                <Card
```
Comment (irenenguyen1017) [inline src/pages/Home.jsx:67]: From line 67 -> 86, you should move to a separate component named `ProductCard`

File: src/pages/Home.jsx [hunk 2]
```diff
@@ -1,9 +1,134 @@
...
+          </div>
+        </div>
+
+        {/* Featured Collection Section */}
+        <div className='flex flex-col mt-10'>
+          <h1 className='text-left font-bold text-3xl mb-4 mt-5 text-primary '>
+            Featured Collection
+          </h1>
+          <div className='carousel mt-4 overflow-hidden'>
+            <motion.div
```
Comment (irenenguyen1017) [inline src/pages/Home.jsx:98]: We also need to move this `motion.div` to new component named `ProductCardList` component to reuse in multiple places.

File: src/pages/Home.jsx [hunk 3]
```diff
@@ -1,9 +1,134 @@
...
+                    <p className='text-default-500'>${product.price}</p>
+                  </CardFooter>
+                </Card>
+              ))}
+            </motion.div>
+          </div>
+        </div>
+
+        {/* Featured Collection Section */}
+        <div className='flex flex-col mt-10'>
```
Comment (irenenguyen1017) [inline src/pages/Home.jsx:93]: We also need to move this  `div` to new component named ProductCardList component to reuse in multiple places.

**Comment by Mana12011207** [comment]
ProductCard and ProductCardList components were created to be re-used on different pages. I also set the number of products displayed in the new arrivals and featured collection to seven. This allows for faster loading of product images. (The previous settings took a lot longer to load images and were not stable).

File: src/components/ProductCard.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Card, CardBody, CardFooter, Image } from '@nextui-org/react';
+
+const ProductCard = ({ product }) => {
+  return (
+    <Card shadow='sm' isPressable className='border-2 w-[300px] h-[450px] flex-shrink-0'>
```
Comment (irenenguyen1017) [inline src/components/ProductCard.jsx:6]: We need to wrap this with `Link` component from `react-router-dom` to enable navigation to `Product` page. We can do like

```javascript
import { Link } from 'react-router-dom';


return (<Link to={`/product/${product.slug}` alt={`link-to-${product.name}`}}>{....}</Link>
```

File: src/components/ProductCardList.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { motion } from 'framer-motion';
+import ProductCard from './ProductCard';
+
+const ProductCardList = ({ products, title }) => {
+  return (
+    <div className='flex flex-col mt-10'>
+      <h1 className='text-left font-bold text-3xl mb-4 mt-5 text-primary '>{title}</h1>
+      <div className='carousel mt-4 overflow-hidden'>
+        {/* Make the 'motion.div' draggable along the horizontal axis and 'left: -500' means the element can be dragged up to 500 pixels to the left of its starting position. */}
+        <motion.div className='flex space-x-2' drag='x' dragConstraints={{ left: -500, right: 0 }}>
```
Comment (irenenguyen1017) [inline src/components/ProductCardList.jsx:10]: We should also the grid layout in desktop view while maintaining the scrolling content in mobile. To achieve that, I think we can try

```javascript
      <div className='flex overflow-x-auto lg:overflow-x-visible lg:grid lg:grid-cols-[repeat(auto-fill,_minmax(240px,_1fr))] gap-8 p-4 no-scrollbar'>
{products.map((product, index) => (
            // key={index} provides a unique key for each ProductCard component
            <ProductCard key={index} product={product} />
          ))}
     </div>


```

**Review by irenenguyen1017 — APPROVED** [review]

**Pull request merged**